zephyr/drivers/can/can_sccan.c

```diff
@@ -125,6 +125,26 @@ LOG_MODULE_REGISTER(sc_can, CONFIG_CAN_LOG_LEVEL);
 #define SCCAN_TXID_EX2(x) ((x & GENMASK(17,0)) << 1)
 #define SCCAN_TXERTR(x)   (x)
 
+/* CAN RX Message Register1 */
+#define SCCAN_RXID1_STD(x) ((x & GENMASK(31,21)) >> 21)
+#define SCCAN_RXID1_EXT(x) ((x & GENMASK(31,21)) >> 3)
+#define SCCAN_RXSRTR(x)    ((x & BIT(20)) >> 20)
```

Comment:
Does it need to be shifted?

Reply:
`SCCAN_RXID1_STD` and `SCCAN_RXID1_EXT` is CAN ID parameter, so need to shift.
However `SCCAN_RXSRTR` is flags, so no need to shift.
Will fix it.